diff -up cups-filters-1.0.34/utils/cups-browsed.c.lookup cups-filters-1.0.34/utils/cups-browsed.c
--- cups-filters-1.0.34/utils/cups-browsed.c.lookup	2013-06-26 16:23:38.000000000 +0200
+++ cups-filters-1.0.34/utils/cups-browsed.c	2013-06-26 16:24:28.491804795 +0200
@@ -990,23 +990,6 @@ found_cups_printer (const char *remote_h
   debug_printf("cups-browsed: browsed queue name is %s\n",
 	       local_resource + 9);
 
-  /* Does the host need resolving? */
-  if (host[strspn (host, "0123456789.")] == '\0') {
-    /* Yes. Resolve it. */
-    struct addrinfo hints, *addr;
-    memset(&hints, 0, sizeof(struct addrinfo));
-    hints.ai_family = AF_UNSPEC;
-    hints.ai_socktype = SOCK_DGRAM;
-    hints.ai_flags = 0;
-    hints.ai_protocol = 0;
-    if (!getaddrinfo (host, NULL, &hints, &addr)) {
-      getnameinfo (addr->ai_addr, addr->ai_addrlen,
-		   host, sizeof(host),
-		   NULL, 0, 0);
-      freeaddrinfo (addr);
-    }
-  }
-
   generate_local_queue(host, port, local_resource, info ? info : "", "", "");
 }
 
